python/torch_mlir/eager_mode/torch_mlir_dispatch.py

```diff
@@ -0,0 +1,258 @@
+    Just as by default we compile for every shape, we also compile for every set of non-tensor inputs.
+
+    Works by
+    1. creating an empty ts graph
+    2. creating and inserting a node corresponding to the op; nodes inserted are just names (hence following steps)
+    2. for each arg to the op:
+        a. if arg is a tensor: add input to the graph corresponding to arg
+           if arg is a constant (bool, int, list): inline a constant (at the top of the graph)
+        b. wire value from previous step to the op (ie SSA value corresponding to graph input is "passed" to op)
+        c. if arg is a tensor: fill in shape
+
+    Parameters
+    ----------
+    schema: torch._C.FunctionSchema (i.e., c10::FunctionSchema)
+        Function in TorchScript IR to turn into MLIR.
+    args: ...
+
+    """
+    graph = torch._C.Graph()
+    node = graph.insertNode(graph.create(schema.name, len(schema.returns)))
+    for i, arg in enumerate(schema.arguments):
+        kwarg = False
+        if arg.name in kwargs:
+            val = kwargs[arg.name]
+            kwarg = True
+        else:
+            val = args[i]
+
+        # if tensor then pass to op
+        if is_tensor_type(arg.type) and val is not None:
+            inp = graph.addInput()
+            if isinstance(arg.type, torch.OptionalType):
+                inp.setType(arg.type.getElementType())
+            else:
+                inp.setType(arg.type)
+
+            if kwarg:
+                inp.setDebugName(arg.name)
+        # if constant then inline
+        else:
+            inp = graph.insertConstant(val)
+            inp.node().moveBefore(node)
+
+        node.addInput(inp)
+
+    if node.hasMultipleOutputs():
+        for outp in node.outputs():
+            graph.registerOutput(outp)
+    else:
+        graph.registerOutput(node.output())
+
+    # this name will ultimately be the name of the func op in the mlir module and thus needs to be unique
+    # per op (since we keep the module around), hence smash together all of the things
+    fn_name = urlify(str(node).strip())
+    fn = torch._C._create_function_from_graph(fn_name, graph)
+    return fn
+
+
+def annotate_args_kwargs(
+    script_fun: torch._C.ScriptFunction,
+    normalized_args: List[Any],
+    normalized_kwargs: Dict[str, Any],
+):
+    unwrapped_normalized_args = tree_map(
+        # TODO: what's the cost here?
+        lambda x: x.detach().contiguous().numpy() if isinstance(x, torch.Tensor) else x,
+        normalized_args,
+    )
+    unwrapped_normalized_kwargs = tree_map(
+        # TODO: what's the cost here?
+        lambda x: x.detach().contiguous().numpy() if isinstance(x, torch.Tensor) else x,
+        normalized_kwargs,
+    )
+
+    annotations = []
+    tensor_args = []
+    for i, arg in enumerate(unwrapped_normalized_args):
+        if isinstance(arg, np.ndarray):
+            # if we get a 0 size dim we need to replace it with symbolic dim
```

Comment:
IIUC, the issue here is that our backends don't support `tensor<0x...` well, right? Can you be more specific about what goes wrong, or why we need this hack?